example/deployment.go

```diff
@@ -32,76 +18,37 @@ func WriteDeploymentFilesExample() error {
 	}
 
 	// Select the deployment type to generate the files for (must correspond to a directory in the template/deployments directory)
-	deploymentType := "manifests"
+	deploymentTemplateType := "deployment-manifests"
 
 	// Create a DraftConfig of inputs to the template (must correspond to the inputs in the template/deployments/<deploymentType>/draft.yaml files)
-	deployConfig := &config.DraftConfig{
-		Variables: []*config.BuilderVar{
-			{
-				Name: "PORT",
-				Default: config.BuilderVarDefault{
-					Value: "80",
-				},
-				Description: "the port exposed in the application",
-				Value:       "8080",
-			},
-			{
-				Name:        "APPNAME",
-				Description: "the name of the application",
-				Value:       "example-app",
-			},
-			{
-				Name: "SERVICEPORT",
-				Default: config.BuilderVarDefault{
-					ReferenceVar: "PORT",
-				},
-				Description: "the port the service uses to make the application accessible from outside the cluster",
-				Value:       "8080",
-			},
-			{
-				Name: "NAMESPACE",
-				Default: config.BuilderVarDefault{
-					Value: "default",
-				},
-				Description: "the name of the image to use in the deployment",
-				Value:       "example-namespace",
-			},
-			{
-				Name: "IMAGENAME",
-				Default: config.BuilderVarDefault{
-					IsPromptDisabled: true,
-					Value:            "the name of the image to use in the deployment",
-				},
-				Description: "the name of the image to use in the deployment",
-				Value:       "example-image",
-			},
-			{
-				Name: "IMAGETAG",
-				Default: config.BuilderVarDefault{
-					IsPromptDisabled: true,
-					Value:            "latest",
-				},
-				Description: "the tag of the image to use in the deployment",
-				Value:       "latest",
-			},
-			{
-				Name: "GENERATORLABEL",
-				Default: config.BuilderVarDefault{
-					IsPromptDisabled: true,
-					Value:            "draft",
-				},
-				Description: "the label to use to identify the deployment as generated by draft",
-			},
-		},
+	templateVars := map[string]string{
+		"PORT":           "8080",
+		"APPNAME":        "example-app",
+		"SERVICEPORT":    "8080",
+		"NAMESPACE":      "example-namespace",
+		"IMAGENAME":      "example-image",
+		"IMAGETAG":       "latest",
+		"GENERATORLABEL": "draft",
 	}
 
 	// Set the output path for the deployment files
 	outputPath := "./"
 
-	// Write the deployment files
-	err := WriteDeploymentFiles(&w, outputPath, deployConfig, deploymentType)
+	// Get the deployment template
+	d, err := handlers.GetTemplate(deploymentTemplateType, "", outputPath, &w)
+	if err != nil || d == nil {
+		return fmt.Errorf("failed to get template: %e", err)
+	}
+
+	// Set the variable values within the template
+	for k, v := range templateVars {
+		d.Config.SetVariable(k, v)
```

Comment:
💡 Suggestion/Observation thought share only please:

```*.go
// Set the variable values within the template
for k, v := range templateVars {
    if err := d.Config.SetVariable(k, v); err != nil {
        return fmt.Errorf("failed to set variable %s: %w", k, err)
    }
}

```

Reply:
`SetVariable` doesnt have a response on it so no error can be thrown